source/game.mission.highscore.bmx

```diff
@@ -0,0 +1,97 @@
+SuperStrict
+
+Import "Dig/base.util.data.bmx"
+?bmxng
+Import "Dig/external/persistence.mod/persistence_mxml.bmx"
+?not bmxng
+Import "Dig/external/persistence.mod/persistence.bmx"
+?
+
+Type TMissionHighscore
+	'TODO how to access the local time for displaying when the game was played
+	'Field realDate:Long
+	Field playerID:Int
+	Field playerName:String
+	Field channelName:String
+	Field gameMinutes:Long
+	Field missionAccomplished:Int = False
+	Field aiPlayer:Int = False
+	Field playerDifficulty:String
+	'encoded entry of the high score, the mission must be capable of translating it back
+	'for presentation and ordering entries
+	Field value:TData
+
+	rem
+	Method isDuplicate:Int(other:TMissionHighscore)
+		If playerID <> other.playerID Then Return False
+		If playerName <> other.playerName Then Return False
+		If channelName <> other.channelName Then Return False
+		If missionAccomplished <> other.missionAccomplished Then Return False
+		If aiPlayer <> other.aiPlayer Then Return False
+		If playerDifficulty <> other.playerDifficulty Then Return False
+		If Abs(gameMinutes - other.gameMinutes) > 1 Then Return False
+		'If Abs(realDate - other.realDate) > 500 Then Return False
+		'TODO compare data?
+		Return True
+	End Method
+	endrem
+End Type
+
+'TODO incorporate number of days per season in missionID (different lists) or highscore itself
+Type TMissionHighscores
+	Field missionID:String
+	Field missiondifficulty:String
+	Field scores:TMissionHighscore[] = new TMissionHighscore[0]
+End Type
+
+Type TAllHighscores
+	Field scores:TMissionHighscores[] = new TMissionHighscores[0]
+	Global lastID:String {noSave}
+	Global lastDifficulty:String {noSave}
+	Global lastScore:TMissionHighscore {noSave}
+
+	Function addEntry(missionID:String, difficulty:String, score:TMissionHighscore)
+		rem
+		'prevent duplicate entries - originally cause be duplicate listener registration
+		If lastID And lastID = missionID And lastDifficulty = difficulty
+			If score.isDuplicate(lastScore) then Return
+		EndIf
+		lastID = missionID
+		lastDifficulty = difficulty
+		lastScore = score
+		endrem
+
+		TPersist.format=True
+		TPersist.maxDepth = 4096
+		Local p:TPersist = New TXMLPersistenceBuilder.Build()
+		'TODO file location and name
+		Local file:String = "highscores.xml"
```

Comment:
Ich denke wir sollten wohl generell irgendwo ein "userfileDir" definieren - was dann in "Eigene Dateien" usw (also OS-abhaengig) umgemuenzt werden kann. Alles was wir "zusaetzlich" speichern, kommt dann von dort (settings.user.xml, highscores, savegames). Gern beim Einladen der DB auch dort ein "userfileDir/database" mit versuchen einzuladen.